src/openrct2-ui/windows/LoadSave.cpp

```diff
@@ -427,694 +471,639 @@ static u8string Browse(bool isSave)
     u8string outPath = ContextOpenCommonFileDialog(desc);
     if (!outPath.empty())
     {
-        // When the given save type was given, Windows still interprets a filename with a dot in its name as a custom extension,
-        // meaning files like "My Coaster v1.2" will not get the .td6 extension by default.
+        // When the given save type was given, Windows still interprets a filename with a dot in its name as a custom
+        // extension, meaning files like "My Coaster v1.2" will not get the .td6 extension by default.
         if (isSave && get_file_extension_type(outPath) != fileType)
             outPath = Path::WithExtension(outPath, extension);
     }
 
     return outPath;
 }
 
-static void WindowLoadsaveMouseup(rct_window* w, WidgetIndex widgetIndex)
+class LoadSaveWindow final : public Window
 {
-    bool isSave = (_type & 0x01) == LOADSAVETYPE_SAVE;
-    switch (widgetIndex)
+public:
+    LoadSaveWindow(int32_t type)
     {
-        case WIDX_CLOSE:
-            window_close(*w);
-            break;
+        widgets = window_loadsave_widgets;
 
-        case WIDX_UP:
-            WindowLoadsavePopulateList(w, isSave, _parentDirectory, _extensionPattern);
-            WindowInitScrollWidgets(*w);
-            w->no_list_items = static_cast<uint16_t>(_listItems.size());
-            break;
+        min_width = WW;
+        min_height = WH / 2;
+        max_width = WW * 2;
+        max_height = WH * 2;
 
-        case WIDX_NEW_FILE:
-            WindowTextInputOpen(
-                w, WIDX_NEW_FILE, STR_NONE, STR_FILEBROWSER_FILE_NAME_PROMPT, {}, STR_STRING,
-                reinterpret_cast<uintptr_t>(_defaultPath.c_str()), 64);
-            break;
+        const auto uiContext = OpenRCT2::GetContext()->GetUiContext();
+        if (!uiContext->HasFilePicker())
+        {
+            disabled_widgets |= (1ULL << WIDX_BROWSE);
+            window_loadsave_widgets[WIDX_BROWSE].type = WindowWidgetType::Empty;
+        }
 
-        case WIDX_NEW_FOLDER:
-            WindowTextInputRawOpen(w, WIDX_NEW_FOLDER, STR_NONE, STR_FILEBROWSER_FOLDER_NAME_PROMPT, {}, "", 64);
-            break;
+        const bool isSave = (type & 0x01) == LOADSAVETYPE_SAVE;
+        const auto path = GetDir(type);
```

Comment:
Maybe these should be parameters of the constructor so we don't have duplicated code

Reply:
Yeah, I was thinking that too, just like with the window class itself now.

Reply:
`type` already is a constructor argument now, `GetDir` returns the default/last directory for a type.